Documentation/security/network/encryption-wireguard.rst

```diff
@@ -216,17 +215,73 @@ have WireGuard encryption enabled, i.e. mixed mode is currently not supported.
 In addition, UDP traffic between nodes of different clusters on port ``51871``
 must be allowed.
 
+Node-to-Node Encryption (beta)
+==============================
+
+By default, WireGuard-based encryption only encrypts traffic between Cilium-managed
+pods. To enable node-to-node encryption, which additionally also encrypts
+node-to-node, pod-to-node and node-to-pod traffic, use the following configuration
+options:
+
+.. tabs::
+
+    .. group-tab:: Cilium CLI
+
+       If you are deploying Cilium with the Cilium CLI, pass the following
+       options:
+
+       .. code-block:: shell-session
+
+          cilium install --encryption wireguard --node-encryption
+
+    .. group-tab:: Helm
+
+       If you are deploying Cilium with Helm by following
+       :ref:`k8s_install_helm`, pass the following options:
+
+       .. parsed-literal::
+
+           helm install cilium |CHART_RELEASE| \\
+             --namespace kube-system \\
+             --set encryption.enabled=true \\
+             --set encryption.type=wireguard \\
+             --set encryption.nodeEncryption=true
+
+.. warning::
+
+  By default, Cilium will disable node-to-node encryption on any node
+  which has the ``node-role.kubernetes.io/control-plane`` label.
+
+  This is done to ensure worker nodes are always able to communicate with the
+  Kubernetes API to update their WireGuard public keys. With node-to-node
+  encryption, the connection to the kube-apiserver would also be encrypted with
+  WireGuard. This creates a bootstrapping problem where the connection used to
+  update the WireGuard public key is itself encrypted with the public key about
+  to be replaced. This is problematic if a node needs to change its public key,
+  for example because it generated a new private key after a reboot or node
+  re-provisioning.
+
+  Therefore, by not encrypting the connection from and to the kube-apiserver
+  host network with WireGuard, we ensure that worker nodes are
+  never accidentally locked out from the control plane. Note that the Kubernetes
+  control-plane itself is usually still encrypted with mTLS and that
+  pod-to-pod traffic for any Cilium-manged pods on the control-plane nodes are
+  also still encrypted via Cilium's WireGuard implementation.
+
+  The label selector for matching the control-plane nodes which shall not
+  participate in node-to-node encryption can be configured using the
+  ``node-encryption-opt-out-labels`` ConfigMap option. You may disable this
+  feature by using an empty label selector. Note that disabling this feature
+  requires you to manually update a node's public key in its corresponding
+  ``CiliumNode`` CRD, as the node itself will be unable to do so itself.
```

Comment:
Reading this paragraph, I found it not entirely clear that `node-encryption-opt-out-labels` defaults to `node-role.kubernetes.io/control-plane`. On my first read, I thought that `node-role.kubernetes.io/control-plane`-labelled nodes could never use node-to-node encryption (OK I missed the “By default” at the beginning, that's my fault), and figured that the `node-encryption-opt-out-labels` was a specifier for _additional_ labels to exclude.

Maybe just me though. Maybe just read this paragraph again, and see if it would make sense to specify that `node-encryption-opt-out-labels` defaults to `node-role.kubernetes.io/control-plane` for the reasons explained above.

Reply:
Thanks for the feedback. Given that you are not the first to struggle with the semantics of the flag, it's probably not just you and the text needs some improvements. I'll try to address that.

Reply:
I've updated the paragraph slightly to hopefully make it a bit more readable. No fundamental changes, but I hope it still provides a bit more clarity now.

Reply:
Thank you, it does read better now